src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/CreateCollectionCommandResolver.java

```diff
@@ -380,18 +412,68 @@ private Integer validateModelAndDimension(
                         "Model name '%s' for provider '%s' is not supported",
                         userConfig.modelName(), userConfig.provider()));
 
-    // TODO: is dimension required? do we still auto populate the dimension?
+    // Handle models with a fixed vector dimension
     if (model.vectorDimension().isPresent()) {
       Integer configVectorDimension = model.vectorDimension().get();
       if (userVectorDimension == null) {
```

Comment:
Should we also allow passing `0` to mark "not specified"? (not sure, just asking)

Reply:
Hmmm, at least for now, if it's not a fixed dimension, `vectorDimension()` will be null and will not be allowed to pass `0`

Reply:
ok.